### PR TITLE
Add BBE for GraphQL Mutations

### PR DESCRIPTION
**Pull request by ThisaruGuruge**
## Purpose
$Subject

Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/1748

### CODE REVIEW

File: examples/graphql-mutations/graphql_mutations.bal [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+service /graphql on new graphql:Listener(4000) {
+
+    // Define a `Person` object in the service
+    private Person person;
+
+    function init() {
+        // Initialize the `person` value
+        self.person = new("Walter White", 51);
+
```
Comment (shafreenAnfar) [inline examples/graphql-mutations/graphql_mutations.bal:11]: Do we need above new lines ?

File: examples/graphql-mutations/graphql_mutations.bal [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    // A resource function represents a field in root `Query` operation
+    resource function get profile() returns Person {
+
```
Comment (shafreenAnfar) [inline examples/graphql-mutations/graphql_mutations.bal:16]: Do we need above new line ? I would rather prefer without it.

File: examples/graphql-mutations/graphql_mutations.bal (hunk 1 above)
Comment (ThisaruGuruge) [reply examples/graphql-mutations/graphql_mutations.bal:11]: To show how we serially change the object, I felt like this is the easiest way. We keep a class-level variable, and we mutate it in two steps. That's why used defined a class variable.

File: examples/graphql-mutations/graphql_mutations.bal (hunk 2 above)
Comment (ThisaruGuruge) [reply examples/graphql-mutations/graphql_mutations.bal:16]: We cannot have a GraphQL service without a `Query` object. Therefore, we must have a resource function in order to be a valid GraphQL service.

**Review by shafreenAnfar — APPROVED** [review]

File: examples/graphql-mutations/graphql_mutations.bal [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    // A remote function represents a field in root `Mutation` operation.
+    // After updating the name, the `person` object will be returned
+    remote function updateName(string name) returns Person {
+
+        self.person.setName(name);
+        return self.person;
+    }
+
+    // Remote function to update the age
```
Comment (praneesha) [inline examples/graphql-mutations/graphql_mutations.bal:28]:
```suggestion
    // Remote function to update the age.
```

File: examples/graphql-mutations/graphql_mutations.bal [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    remote function updateAge(int age) returns Person {
+
+        self.person.setAge(age);
+        return self.person;
+    }
+}
+
+// Define a service class to use as an object in GraphQL service.
```
Comment (praneesha) [inline examples/graphql-mutations/graphql_mutations.bal:36]:
```suggestion
// Define a service class to use as an object in the GraphQL service.
```

File: examples/graphql-mutations/graphql_mutations.client.out [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# Send a query to the GraphQL endpoint using a cURL command.
+# The query used: { profile { name isAdult } }
+curl -X POST -H "Content-type: application/json" -d '{ "query": "{ profile { name age } }" }' 'http://localhost:4000/graphql'
+{"data":{"profile":{"name":"Walter White", "age":51}}}
+
+# Now send a mutation to update the name
```
Comment (praneesha) [inline examples/graphql-mutations/graphql_mutations.client.out:6]:
```suggestion
# Now send a mutation to update the name.
```

File: examples/graphql-mutations/graphql_mutations.client.out [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+# The document used for this: mutation { updateName(name: \"Mr. Lambert\") { name age } }
+curl -X POST -H "Content-type: application/json" -d '{ "query": "mutation { updateName(name: \"Mr. Lambert\") { name age } }" }' 'http://localhost:4000/graphql'
+{"data":{"updateName":{"name":"Mr. Lambert", "age":51}}}
+
+# Then send a mutation to update the age
```
Comment (praneesha) [inline examples/graphql-mutations/graphql_mutations.client.out:11]:
```suggestion
# Then send a mutation to update the age.
```

File: examples/graphql-mutations/graphql_mutations.description [hunk 7]
```diff
@@ -0,0 +1,9 @@
+// A remote function inside a GraphQL service represents a field in the root
+// `Mutation` object type. Therefore, if a remote function is present inside
+// the Ballerina GraphQL service, the auto-generated schema will have a
+// `Mutation` type. Each remote function in the service will be added as a
+// field of the `Mutation` type. The field name will be the remote function
+// name, and the field type will the the return type of the remote function.
```
Comment (praneesha) [inline examples/graphql-mutations/graphql_mutations.description:6]:
```suggestion
// name and the field type will be the return type of the remote function.
```

File: examples/graphql-returning-service-objects/graphql_returning_service_objects.bal [hunk 8]
```diff
@@ -11,7 +11,7 @@ service /graphql on new graphql:Listener(4000) {
     }
 }
 
-// Define a service class to use in GraphQL service.
+// Define a service class to use as an object in GraphQL service.
```
Comment (praneesha) [inline examples/graphql-returning-service-objects/graphql_returning_service_objects.bal:14]:
```suggestion
// Define a service class to use as an object in the GraphQL service.
```

**Review by praneesha — APPROVED** [review]

**Pull request merged**